tools/logger/convert.c

```diff
@@ -131,14 +131,48 @@ static const char *format_uid(uint32_t uid_ptr, int use_colors, bool be, bool up
 	return str;
 }
 
+/* fmt should point '%pUx`, return length of UUID format string or zero */
+static int get_uuid_format(const char *fmt, const char *fmt_end, int *be, int *upper)
```

Comment:
I'd like snprintf semantics: You write as much as you can and return the length of the would-be UUID string. The caller is responsible for reallocating if necessary.

Reply:
changed to `asprintf_uuid`.